tests/Testcontainers.Tests/Fixtures/Containers/Unix/ProtectDockerDaemonSocket.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace DotNet.Testcontainers.Tests.Fixtures
+{
+  using System;
+  using System.IO;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Builders;
+  using DotNet.Testcontainers.Configurations;
+  using DotNet.Testcontainers.Containers;
+  using DotNet.Testcontainers.Images;
+  using Xunit;
+
+  public abstract class ProtectDockerDaemonSocket : IAsyncLifetime
+  {
+    public const string DockerVersion = "20.10.18";
+
+    private const string CertsDirectoryName = "certs";
+
+    private const ushort TlsPort = 2376;
+
+    private readonly ITestcontainersContainer container;
+
+    private readonly IDockerImage image = new DockerImage(string.Empty, "docker", DockerVersion + "-dind");
+
+    public ProtectDockerDaemonSocket(ITestcontainersBuilder<TestcontainersContainer> containerConfiguration)
```

Comment:
I split the fixtures into two classes `DockerMTlsFixture` and `DockerTlsFixture`. We can configure the build individual for each of them. I noticed that the env variables do not have any effect on the container configuration / the Docker daemon, so I removed them.

Reply:
I merged them since they are too many common parts. But it is ok.